src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -120,8 +120,9 @@ public JsonWebToken(string header, string payload)
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14302, payload), ex));
             }
 
-            _encodedHeader = header;
-            _encodedPayload = payload;
+            _encodedHeader = Base64UrlEncoder.Encode(header);
+            _encodedPayload = Base64UrlEncoder.Encode(payload);
```

Comment:
In the EncodedPayload property, the _encodedPayload is set to:

`_encodedPayload = IsEncrypted ? string.Empty : EncodedToken.Substring(Dot1 + 1, Dot2 - Dot1 - 1)`